src/bin/kafka_consumer.rs

```diff
@@ -31,44 +33,62 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     let args = Cli::parse();
 
-    // TODO: based on the location of the telescope, figure out the exact timestamp
-    // for the start of the night
     let date = match args.date {
         Some(date) => chrono::NaiveDate::parse_from_str(&date, "%Y%m%d").unwrap(),
         None => chrono::Utc::now().date_naive().pred_opt().unwrap(),
     };
-    let survey = args.survey;
+    let date = date.and_hms_opt(0, 0, 0).unwrap();
+    let timestamp = chrono::Utc.from_utc_datetime(&date).timestamp();
+
+    let program_id = match args.program_id {
```

Comment:
Would this be better as an enum?

Reply:
Yes let me clean up all those binaries with proper enums where applicable

Reply:
done